### ISSUE TITLE
Add Guidance for Feel expressions in Cloud diagrams

### ISSUE THREAD

**Issue by marstamm** [bpmn-io/bpmn-js-properties-panel#584]
__What should we do?__
In the first Iteration, we should

- [x] #599
- [x] #600
- [x] #606
    - [x] Add marker to input fields  mapped through element templates
    - [x] Add ability to mark element template fields as FEEL enabled (despite not being a FEEL enabled input field; example: Header that is processed with FEEL scripting enabled downstream) 
- [x] Update [docs](https://docs.camunda.io/docs/components/modeler/desktop-modeler/element-templates/defining-templates/) - https://github.com/camunda-cloud/camunda-cloud-documentation/pull/699
- [x] Add example template to https://github.com/camunda/camunda-modeler/blob/develop/resources/element-templates/cloud-samples.json

__Why should we do it?__
Those are low hanging fruits to make it easier to use FEEL in Camunda Cloud

https://github.com/camunda/product-hub/issues/4

**Comment by nikku** [comment]
Discussed with @marstamm 2022-03-01:

* If we provide a visual indicator for FEEL editing, we shall provide it in fields mapped via [element templates](https://github.com/camunda/camunda-modeler/tree/master/docs/element-templates#readme), too

* As we embrace FEEL for scripting we shall make that indicator overridable via the element template definition. Use case: A `zeebe:header` that is later interpreted as FEEL by a task run-time. For example, we could mark an element template field as `{ ...feel: true || false || "maybe" }`: 
    ![image](https://user-images.githubusercontent.com/58601/156319924-cd98d0df-f83f-42d0-969a-c88f55b8274a.png)

* The visual indicator for FEEL editing shall work for both text inputs as well as `textarea` (script editing case):
    
![image](https://user-images.githubusercontent.com/58601/156321941-9006de72-bbef-4b3c-b51e-baf772e50917.png)

**Comment by MaxTru** [comment]
Alternative to consider: 
* FEEL input is just another input type. So rather than adding a new config option `feel`, consider to just add a `type`: https://github.com/camunda/camunda-modeler/tree/develop/docs/element-templates#types
* As-is: `String`, `Text`, `Boolean`, `Dropdown`, `Hidden`
* New: `String`, `Text`, `Boolean`, `Dropdown`, `Hidden`, `FeelString`, `FeelText` (or so)
* This would work around situations where we need to validate / handle situations like: 
  * User defines type `boolean`
  * User sets `feel: true`
  * => does not make sense

**Comment by MaxTru** [comment]
cc @pinussilvestrus

**Comment by marstamm** [comment]
> `FeelString`, `FeelText`

We would still need some way to track the option "Feel is optional/ Feel is required". If we go with input type only, that would mean 4 new input fields (`FeelString`, `FeelText`, `OptionalFeelString`, `OptionalFeelText`). I will reach out to @pinussilvestrus to refine our approach

**Comment by pinussilvestrus** [comment]
> * This would work around situations where we need to validate/handle situations like:

I think this is something our JSON schema can handle quite easily by defining rules 🙂 In general, I'd avoid adding new `types`, but making the element template properties panel entries (aka `CustomProps` ) reactive to the attributes you can define in your template. Just the way we did with `optional` => the custom props reads this attribute and react to it, no need to provide a type like "OptionalString" or something else.

Without going too much into detail, I'd find a simple attribute like `feel` pretty convenient, from a configuration perspective. But I think we will find out 👍

**Comment by MaxTru** [comment]
> > * This would work around situations where we need to validate/handle situations like:
> 
> I think this is something our[ JSON schema](https://github.com/camunda/element-templates-json-schema) can handle quite easily by defining rules slightly_smiling_face In general, I'd avoid adding new `types`, but making the element template properties panel entries (aka `CustomProps` ) reactive to the attributes you can define in your template. Just the way we did with `optional` => the custom props reads this attribute and react to it, no need to provide a type like "OptionalString" or something else.
> 
> Without going too much into detail, I'd find a simple attribute like `feel` pretty convenient, from a configuration perspective. But I think we will find out +1

Sure. The alternative I suggested came into my mind, because logically / conceptually for me it works as following:
* `type` defines the type of the input element
* So if you put `type = checkbox` you will have a checkbox
* If you put `type = string` you will have a simple single line input
* And now if you put `type = feelString` you will have a input, which has some additional markers (or whatever) to ensure that only FEEL is put in.

As said, I am also okay with introducing a new config option `feel`, but I think having a dedicated type is also worth considering.

**Comment by marstamm** [comment]
Decision regarding design: we will have the Icon as part of the label, which is also how Carbon does it. For Hover, we will go with browser-default using the `title` tag for now, might refine later:

![image](https://user-images.githubusercontent.com/21984219/156529309-80047c83-a117-4ddf-8e9d-60298dd2aba8.png)

**Comment by marstamm** [comment]
Reopening due to missing examples in the Modeler Repository

**Comment by marstamm** [comment]
Done with https://github.com/camunda/camunda-modeler/pull/2822

**Comment by nikku** [comment]
Is this released already?

**Comment by MaxTru** [comment]
> Is this released already?

Yes [alpha.6](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/CHANGELOG.md#100-alpha6)

**Comment by marstamm** [comment]
Yes, this is ready to be released (and already part of [1.0.0-alpha.6](https://github.com/bpmn-io/bpmn-js-properties-panel/commit/7a7992417a0175300d849acc99b3bfbfed367e4c))

**Comment by nikku** [comment]
[Pretty cool](https://camunda.slack.com/archives/C02JLRNQQ05/p1648023374247189).
